src/actions/arcActions.ts

```diff
@@ -406,14 +406,26 @@ async function getRedemptions(avatarAddress: string, votingMachineInstance: Arc.
     voterTokens: Util.fromWei(await votingMachineInstance.getRedeemableTokensVoter({ proposalId, beneficiaryAddress: accountAddress })).toNumber(),
   };
 
+  // Beneficiary rewards
   if (proposal.beneficiaryAddress == accountAddress) {
-    redemptions.beneficiaryEth = (await proposalInstance.contract.getPeriodsToPay(proposalId, avatarAddress, ContributionRewardType.Eth)) * proposal.ethReward;
-    redemptions.beneficiaryNativeToken = (await proposalInstance.contract.getPeriodsToPay(proposalId, avatarAddress, ContributionRewardType.NativeToken)) * proposal.nativeTokenReward;
-    redemptions.beneficiaryReputation = (await proposalInstance.contract.getPeriodsToPay(proposalId, avatarAddress, ContributionRewardType.Reputation)) * proposal.reputationChange;
+    if (proposal.state == ProposalStates.Boosted && proposal.winningVote === VoteOptions.Yes) {
+      // Boosted proposal that passed by expiring with more yes votes than no
+      //   have to manually calculate beneficiary rewards
+      const propDetails = await proposalInstance.contract.organizationsProposals(avatarAddress, proposalId);
```

Comment:
You can do this using `(await proposalInstance.getDaoProposals({ proposalId }))[0].numberOfPeriods

Reply:
ah cool